### PR TITLE
IE11 Ajax redirect 308 fixes #17434

### LINKED ISSUE #17434: Redirecting problem in Internet Explorer 11 (IE11)
### What steps will reproduce the problem?
Performing an ajax request and doing a redirect from it.
(Issue when using Internet Explorer 11 (IE11))
Related (same issue) #9670 
### What is the expected result?
A successful redirect to given URL in function Response::redirect()
### What do you get instead?
No redirect!
jokeie11

### Additional info
Using 308 statusCode (default is 302) solves the problem in general but it is probably not the right way to do this.
`return Yii::$app->getResponse()->redirect(['aaa/view', 'id' => $this->lid], 308);`

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.22
| PHP version      |  5.6.40
| Operating system | W10x64

### PR DESCRIPTION
**Pull request by kamarton**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #17434

### CODE REVIEW

File: framework/web/Response.php [hunk 1]
```diff
@@ -857,18 +857,22 @@ public function redirect($url, $statusCode = 302, $checkAjax = true)
...
         if ($checkAjax) {
-            if (Yii::$app->getRequest()->getIsAjax()) {
-                if (Yii::$app->getRequest()->getHeaders()->get('X-Ie-Redirect-Compatibility') !== null && $statusCode === 302) {
+            if ($request->getIsAjax()) {
+                $is302 = $statusCode === 302;
+                if ($is302 && $request->getHeaders()->get('X-Ie-Redirect-Compatibility') !== null) {
                     // Ajax 302 redirect in IE does not work. Change status code to 200. See https://github.com/yiisoft/yii2/issues/9670
                     $statusCode = 200;
+                } elseif ($is302 && $request->isAjax && $request->isPost && preg_match('/Trident.*\brv\:11\./' /* IE11 */, $request->userAgent)) {
+                    $statusCode = 308;
```
Comment (samdark) [inline framework/web/Response.php]: Would that work in case `X-Ie-Redirect-Compatibility` present?

Comment (kamarton) [reply framework/web/Response.php]: Good idea, I'll test it more.

Comment (kamarton) [reply framework/web/Response.php]: Notes for me:
- https://github.com/yiisoft/jquery-pjax/issues/37
- https://github.com/yiisoft/jquery-pjax/releases/tag/v2.0.6
- https://github.com/yiisoft/jquery-pjax/pull/38

Comment (kamarton) [reply framework/web/Response.php]: As I look at the code the two are the same. `X-Ie-Redirect-Compatibility` header is restricted to only pjax.
I'll think better about it and test it, but probably the `X-Ie-Redirect-Compatibility` header header condition is not needed.

Comment (kamarton) [reply framework/web/Response.php]: | check list for redirects | result | 
| ------------ |----|
| pjax         |     |
| pjax + POST + IE 11 |    |
| pjax + GET + IE 11   |    |
| pjax + POST + not IE 11 |    |
| pjax + GET + not IE 11 |    |
| ajax |     |
| ajax + POST + IE 11 |    |
| ajax + GET + IE 11   |    |
| ajax + POST + not IE 11 |    |
| ajax + GET + not IE 11 |    |

Comment (kamarton) [reply framework/web/Response.php]: | check list for http codes | ie11 result |
|---|---|  
|301| |
|302| ✔️ |
|308| |

Comment (kamarton) [reply framework/web/Response.php]: @samdark reviewable / testable

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Thank you!

**Pull request merged**